profiling/simple-kernel-timer/kp_reader.cpp

```diff
@@ -64,8 +64,9 @@ int main(int argc, char* argv[]) {
       KernelPerformanceInfo* new_kernel =
           new KernelPerformanceInfo("", PARALLEL_FOR);
       if (new_kernel->readFromFile(the_file)) {
-        if (strlen(new_kernel->getName()) > 0) {
-          int kernelIndex = find_index(kernelInfo, new_kernel->getName());
+        if (!new_kernel->getName().empty()) {
+          int kernelIndex =
+              find_index(kernelInfo, new_kernel->getName().c_str());
```

Comment:
This is silly, please update find_index and take the trailing argument as a string const ref.

Reply:
Indeed, and done.

By the way, I think `find_index` is useless if the variable `kernelInfo`, which is a `std::vector`, is changed to being a `std::set` whose comparator compares the time (see what `compareKernelPerformanceInfo` does).

If you agree I can make the change.

Reply:
@dalg24 I guess you'll want to make the change in another PR :wink: